packages/react-testing/src/root.tsx

```diff
@@ -140,10 +141,14 @@ export class Root<Props> implements Node<Props> {
           return result as unknown as Promise<void>;
         }
 
-        return Promise.race([
+        const promise = Promise.race([
           result,
           this.actPromise,
         ]) as unknown as Promise<void>;
+
+        this.actPromises.push(promise);
```

Comment:
We collect all the promises we need to wait on as `Promise.race([Promise.resolve()]` does not resolve at the same time as `Promise.resolve()` (it adds a tick). So best to wait on the same promise that gets returned to `act`